#     Profitable Apps in Google Play Store and                                          Apple Store

This is a project to analyse various apps data from Google Play Store and Apple store.

The goal of this project is to analyse and better understand the user interaction with different apps in Google Play Store and Apple Store.

Let's start by extracting datasets from 'AppleStore.csv' and 'googleplaystore.csv' files that we have previously downloaded from these links: [Google apps data](https://www.kaggle.com/lava18/google-play-store-apps/home) and [Apple apps data](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/home).

In [1]:
from csv import reader

# Extracting Apple apps dataset
open_apple = open('AppleStore.csv', encoding='utf8')
read_apple = reader(open_apple)
ios = list(read_apple)
for row in ios:
    del row[0]
ios_header = ios[0]
ios = ios[1:]

# Extracting Google apps dataset
open_google = open('googleplaystore.csv', encoding='utf8')
read_google = reader(open_google)
android = list(read_google)
android_header = android[0]
android = android[1:]

Now let's create a function to explore different rows and to also count the number of rows and columns from these datasets:

In [2]:
def explore(dataset, start, end, rows_and_column_total = False):
    data_slice = dataset[start:end]
    for row in data_slice:
        print(row)
        print('\n')
    if rows_and_column_total:
        print('Tolal number of rows: ', len(dataset))
        print('Total number of columns: ', len(dataset[0]))

Now let's use this `explore()` function to explore some of the rows of our ios and android dataset:

In [3]:
# Exploring Apple apps dataset
explore(ios, 0, 3, True)

['281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1']


['281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


['281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '188583', '2822', '3.5', '4.5', '5.0.0', '4+', 'Weather', '37', '5', '3', '1']


Tolal number of rows:  7197
Total number of columns:  16


In [4]:
# Exploring Google apps dataset
explore(android, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Tolal number of rows:  10841
Total number of columns:  13


We will need to look at the columns names of these datasets to identify the colomns that might help us with our analysis. For that we are going to print the headers of each dataset:

In [5]:
print(ios_header)
print('\n')
print(android_header)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


There are 7197 apps in the ios datasets and 10841 apps in the android dataset. The columns that look interesting for our analysis are-

- For ios: 'track_name', 'currency', 'price', 'rating_count_tot', 'rating_count_ver' and 'prime_genre'.

- For android: 'App', 'Category', 'Reviews', 'Installs', 'Type' and 'Genre'.

If the column names don't appear to be descriptive enough for us to understand properly we can always go to these links ([Google apps data](https://www.kaggle.com/lava18/google-play-store-apps/home) and [Apple apps data](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/home)) to have a better understanding of these columns.

In Google Play dataset [discussion section](https://www.kaggle.com/lava18/google-play-store-apps/discussion) there is [one of the discussions](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015) describes an error for the row 10472. Let's checkout the row to see if there is any real error by printing out `android_header` and `android[10472]`- 

In [6]:
print(android_header)
print(android[10472])

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


We can see that in the row 10472 of our Google Play dataset information are missing for two columns ('Category' and 'Genre'). Now that we have confirmed that there are indeed two errors, we are going to remove the row from our dataset-

In [7]:
del android[10472]

When we are doing data cleaning, we also need to make sure there is no duplicated data. By exploring the [discussion section](https://www.kaggle.com/lava18/google-play-store-apps/discussion) we have came to know that there are duplicate entries for different apps in our android dataset. For example, 'Instagram' has four different entries-

In [8]:
for row in android:
    app_name = row[0]
    if app_name == 'Instagram':
        print(row)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


There are actually a large number of duplicates in our dataset. Let's find out this number for android dataset-

In [9]:
unique_apps = []
duplicate_apps = []
for row in android:
    app_name = row[0]
    if app_name in unique_apps:
        duplicate_apps.append(app_name)
    else:
        unique_apps.append(app_name)
print('Number of unique app entries: ', len(unique_apps))
print('Number of duplicate app entries: ', len(duplicate_apps))
print('\n')
print('Example of duplicated apps: ', duplicate_apps[:15])

Number of unique app entries:  9659
Number of duplicate app entries:  1181


Example of duplicated apps:  ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


From the above result we can see that there are 1181 duplicate app rows in android dataset.

Now, for effective and accurate data analysis we need to remove all these duplicate entries. We can try removing these duplicates randomly, but we can try to find a better way to do that. For example: if we look at the 4 duplicated 'Instagram' row entries above, we can see that amongst the 4 entries the only difference is happening in  the 4th position (column) where it shows the number of app review. The different number of each entry indicate that those data were collected in different time set. The least number indicates the oldest data collection, while the highest number indicates the latest data collection. We can consider these numbers of app reviews when we are removing duplicate app rows. So instead of randomly removing duplicated rows, we can choose to keep the row with the highest number of app reviews as it indicates being the latest data collection and remove the rest of the duplicates.

To do that we will start by creating an empty dictionary and then populate the dictionary with `key:value` pairs where the key will be a unique app name and the value will be the highest number of that app reviews- 

In [10]:
reviews_max = {}

for row in android:
    name = row[0]
    n_reviews = float(row[3])
    if name not in reviews_max:
        reviews_max[name] = n_reviews
    elif name in reviews_max and n_reviews > reviews_max[name]:
        reviews_max[name] = n_reviews
print(len(reviews_max))

9659


Above we have created the dictionary `reviews_max` and stored all the unique app names alongside their highest number of reviews as `key:value` pair. When we printed out the length of the dictionary, we see that there are 9659 entries of unique apps.

Now we are going to use the dictionary `reviews_max` to remove the duplicate app rows from our android dataset-

In [11]:
android_clean = []
already_added = []

for row in android:
    name = row[0]
    n_reviews = float(row[3])
    if n_reviews == reviews_max[name] and (name not in already_added):
        android_clean.append(row)
        already_added.append(name)

Above we created two different empty lists. `android_clean` to store new cleaned dataset in it and `already_added` to store the name of apps whose rows have already been stored in `android_clean`. Then we looped through the android dataset and for each iteration we assaigned the app name to a variable named `name` and converted the number of reviews to `float` and assigned the value to variable named `n_reviews`. After that we put our condition, if `n_reviews` is equal to the value of same app name in the dictionary `reviews_max` and `name` is not in `alreday_added` then append the entire row to the list `android_clean` and also append the `name` to the list `already_added`.

This way we got our new clean dataset `android_clean`. Just to make sure our code is working properly, we printed the length of our clean dataset and we can see that the output is 9659 which is the same lentgh as our unique apps dictionary `reviews_max`.

Now lets explore the `android_clean` dataset using our `explore()` function to check the number of rows is 9659:

In [12]:
explore(android_clean, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Tolal number of rows:  9659
Total number of columns:  13


Yes we have 9659 rows as expected.

Now lets write a function that will take a string as an input and then return `True` if the string is in English or `False` if the string is in any other language:

In [13]:
def is_english(string):
    for character in string:
        if ord(character) > 127:
            return False
    return True

Above we have use `ord()` function in our `is_english()` function which gives us the corresponding number associated with each character in a string. There are corresponding number associated with all the alphabets from all different languages and for all digits from 0 to 9 (converted to string), punctuation marks (., !, ?, ;)(as a string), and other symbols(+, /)(as a string). According to [ASCII](https://en.wikipedia.org/wiki/ASCII), the numbers corresponding to the characters we commonly use in an English text are all in the range 0 to 127. So we have used `ord()` function, after we iterated through the given string that has been put as an input, to calculate the corresponding number of the iterated string. And if the corresponding number is greater than 127 then it returns `False`, otherwise it returns `True`. This way we can determine if the input string is in English or not.

To check if our `is_english()` function is working properly, we are going put four different strings as input and see the result:

In [14]:
print(is_english('Instagram'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))

True
False
False
False


The function seems to work. But we can see that it has returned `False` for last two strings ('Docs To Go™ Free Office Suite' and 'Instachat 😜') which are actually in English. The problems are the emoji (😜) and the symbol (™). The corresponding numbers for emojis and other symbols fall outside of the ASCII range for the commonly used English texts (127). Because of this, we might will remove useful apps if we use the function `is_english` in its current form.

So to avoid removing English apps with emojis and symbols we will try to change our function `is_english()` in way that it will return `False` if the input string has more than three characters that fall outside the ASCII range of commonly used English text (0 - 127). Otherwise it should return `True`. Lets change our function:

In [15]:
def is_english(string):
    non_english = []
    for character in string:
        if ord(character) > 127:
            non_english.append(character)
    if len(non_english) > 3:
        return False
    else:
        return True

print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))

True
True
False


It seems like our edited function is now working well. Now lets use our new function to filter out non_english apps from both android and ios datasets:

In [16]:
android_english = []
ios_english = []

# lets filter the android_clean dataset
for row in android_clean:
    if is_english(row[0]) == True:
        android_english.append(row)

# And now the ios dataset
for row in ios:
    if is_english(row[1]) == True:
        ios_english.append(row)

explore(android_english, 0, 3, True)
explore(ios_english, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Tolal number of rows:  9614
Total number of columns:  13
['281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1']


['281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


['281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '188583

Above we can see that we have remaining 9614 apps in `android_english` dataset and 6183 apps in `ios_english` dataset.

We analyzing these datasets from google play and apple store so that these insights might help us to build some great apps that will be popular to our users and also we aim to make these apps 'free' to download and install. Our main source of revenue will come from in-app ads.

In the datasets `android_english` and `ios_english` we have both free and paid apps. As we plan to build free apps, we will now try to separate all the free apps from both datasets and store them in separate lists: 

In [17]:
android_final = []
ios_final = []

for row in android_english:
    if row[7] == '0':
        android_final.append(row)

for row in ios_english:
    if row[4] == '0':
        ios_final.append(row)
        
print(len(android_final))
print(len(ios_final))

8864
3222


Now as we can see above, we are left with 8864 android apps and 3222 ios apps which are both free and English apps. `android_final` and `ios_final` are our final datasets for data analysis.

 As we have already mentioned at the begining that our goal is to build free apps targeting an English speaking audience and we will be generating our revenues from in-app ads, we need to now find out which types of apps or apps from which genre attracts maximum users. Cause more users means more revenues.

After our initial data analysis, we will develop an app for google play store and will observe what kind of response it gets from users over next six months. If the app does well then we will develop the app further and also we will develop the IOS version of the app.

So to begin our analysis lets create a frequency table for `prime-genre` column of IOS dataset and for `genres` and `category` columns of Android dataset to get sense of the most common genres for each market.

Now we will build two functions that we can use to analyse the frequency tables:
 - One function to generate frequency tables that show percentages.
 - another one to display the percentages in descending order.

In [18]:
def freq_table(dataset, index):
    table = {}
    total = 0
    for row in dataset:
        total += 1
        value = row[index]
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
    percentage_table = {}
    for key in table:
        percentage = (table[key] / total) * 100
        percentage_table[key] = percentage
    return percentage_table

def display_table(dataset, index):
    table = freq_table(dataset, index)
    display_table = []
    for key in table:
        key_value_tuple = (table[key], key)
        display_table.append(key_value_tuple)
    sorted_table = sorted(display_table, reverse = True)
    for entry in sorted_table:
        print(entry[1], ':', entry[0])

We are going to start by examining the frequency table for `prime_genre` column of IOS dataset. 

In [19]:
display_table(ios_final, -5) #prime genre

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


We can see that in Apple App Store, 58.16% of the apps are of Games genre, 7.88% are of Entertainment, 4.97% are of Photo & Video, 3.66% are of Education, 3.29% are of Social Networking and so on.

From above analysation we can say that, Apple App Store is dominated by fun apps (games, entertainment, photo, social networking, music, etc), while apps with practical purposes (education, medical, shopping, utilities, etc) are not as numerous. However, a large number of fun apps doesn't actually mean that they have a large number of users as well.

Anyway, lets continue by examining `genres` and `category` columns of the Android dataset.

In [20]:
display_table(android_final, 1) #category

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

It shows a different scenario on Google play store. We don't see a lot of fun apps here and practical purposes apps (family, tools, business, lifestyle, productivity, etc) are dominating in numbers. However, the family category, which is accounted for almost 18.91% of the apps, are mostly games for kids.

Now lets have a look at the genre column.

In [21]:
display_table(android_final, -4) #genre

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

Practical apps seem to have better representation on Google Play Store in compared to Apple App Store. And the frequency table of the genre column of Android dataset only confirms the speculation.

However, the genre column is much more granular as it has more categories. As we at this moment we are just looking for the bigger picture, we will work only with the category column from Android dataset for this project.

Up until this point, we have found out that Apple App Store is dominated by fun apps while Google Play Store is more balanced in terms of the number of both fun and practical purposes apps.

Now we will try to find out which types of apps has more users.

# Most Popular Apps in Apple Store by Genre

One way to find out the most popular apps in our datasets is to find out which apps have got installed most. For that we can use the `Installs` column from Our Android dataset. But there is no such column in our IOS dataset. So for IOS dataset we will try to find out which apps have got the most number of user ratings and we can find that in `rating_count_tot` column.

Below we will try to find out the average number of user ratings for each app genre on the IOS app store.

In [22]:
ios_genres = freq_table(ios_final, -5) # app[-5] is the genre column
for genre in ios_genres:
    total = 0
    len_genre = 0
    for app in ios_final:
        app_genre = app[-5]
        num_rating = app[5] # app[5] is the numnber of ratings column
        if app_genre == genre:
            total += float(num_rating)
            len_genre += 1
    genre_avg = total / len_genre
    print(genre, ':', genre_avg)

Productivity : 21028.410714285714
Weather : 52279.892857142855
Shopping : 26919.690476190477
Reference : 74942.11111111111
Finance : 31467.944444444445
Music : 57326.530303030304
Utilities : 18684.456790123455
Travel : 28243.8
Social Networking : 71548.34905660378
Sports : 23008.898550724636
Health & Fitness : 23298.015384615384
Games : 22788.6696905016
Food & Drink : 33333.92307692308
News : 21248.023255813954
Book : 39758.5
Photo & Video : 28441.54375
Entertainment : 14029.830708661417
Business : 7491.117647058823
Lifestyle : 16485.764705882353
Education : 7003.983050847458
Navigation : 86090.33333333333
Medical : 612.0
Catalogs : 4004.0


We can see that navigation apps have the highest number of user ratings. But also, this high number of ratings is influenced by the large volume of user ratings from apps like Waze and Google map.

In [23]:
for app in ios_final:
    if app[-5] == 'Navigation':
        print(app[1], ':', app[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Geocaching® : 12811
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5
CoPilot GPS – Car Navigation & Offline Maps : 3582
Google Maps - Navigation & Transit : 154911


Unfortunately, the same pattern applies for social networking apps as well, where the average rating number is hugely influenced by giant social apps like, Facebook, Twitter, Instagram, etc. The same applies to the music apps, where apps like Pandora, Spotify, Shazam, etc played a big roll for higer avegare rating number.

Our mission is to find the popular genres. But navigation, social networking or music apps seems like more popular than they actually are. If we filter out these giants apps from each app genre and then calculate the average rating we might get a better result.

Lets have a look at our Reference app genre.

In [24]:
for app in ios_final:
    if app[-5] == 'Reference':
        print(app[1], ':', app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
Merriam-Webster Dictionary : 16849
Google Translate : 26786
Night Sky : 12122
WWDC : 762
Jishokun-Japanese English Dictionary & Translator : 0
教えて!goo : 0
VPN Express : 14
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Real Bike Traffic Rider Virtual Reality Glasses : 8


Reference apps have 74,942 ratings. But again we can see that apps like Bible and Dictionary.com have heavily influenced the average rating number.

However, this niche seems to show some potential. One thing we could do is take another popular book and turn it into an app where we could add different features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes about the book, etc. On top of that, we could also embed a dictionary within the app, so users don't need to exit our app to look up words in an external app.

This idea seems to fit well with the fact that the App Store is dominated by for-fun apps. This suggests the market might be a bit saturated with for-fun apps, which means a practical app might have more of a chance to stand out among the huge number of apps on the App Store.

Now lets analyze our Android dataset for Google play market.

# Most Popular Apps in Google Play Store by Genre

In our Android dataset, we actually have the information about the number of installation for each app, which is agreat key for our analysation. However, if we have a look we will see that the numbers provided for installation are open-ended (for example: 100+, 200+, 1000+, etc).

In [25]:
display_table(android_final, 5) # 5 is the index for installs column

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


The problem with this data is it not being precise. The apps with 100,000+ installs could have 200,000 installs or 300,000 installs or so on. But fortunately, we don't have to be precise. As all we need is to get the idea of the apps and genres that attract most users.

We are going to leave the numbers of installation as they are. That means, we will consider apps with 100,000+ installation have 100,000 installation, 200,000+ installation have 200,000 installation and so on. however, we will have to convert the installs numbers to floats. So we will have to remove all the commas and plus characters from them. Otherwise, it will raise errors. We will do that in the loop below.

In [26]:
cat_android = freq_table(android_final, 1) # 1 is the index for Category column
for category in cat_android:
    total = 0
    len_cat = 0
    for app in android_final:
        if category == app[1]:
            n_install = app[5] # app[5] is the installs column
            n_install = n_install.replace(',', '')
            n_install = n_install.replace('+', '')
            total += float(n_install)
            len_cat += 1
    avg_install = total / len_cat
    print(category, ':', avg_install)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

We can see that, apps in Communication genre have the most installs. Approximately 38,456,119 installs. But again, this number has been greatly influenced by some giant apps with over a billion installation (Whatsapp, Facebook Messanger, Skype, Gmail, etc).

In [27]:
for app in android_final:
    if app[1] == 'COMMUNICATION':
        print(app[0], ':', app[5]) # app[0] is the application name column

WhatsApp Messenger : 1,000,000,000+
Messenger for SMS : 10,000,000+
My Tele2 : 5,000,000+
imo beta free calls and text : 100,000,000+
Contacts : 50,000,000+
Call Free – Free Call : 5,000,000+
Web Browser & Explorer : 5,000,000+
Browser 4G : 10,000,000+
MegaFon Dashboard : 10,000,000+
ZenUI Dialer & Contacts : 10,000,000+
Cricket Visual Voicemail : 10,000,000+
TracFone My Account : 1,000,000+
Xperia Link™ : 10,000,000+
TouchPal Keyboard - Fun Emoji & Android Keyboard : 10,000,000+
Skype Lite - Free Video Call & Chat : 5,000,000+
My magenta : 1,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Seznam.cz : 1,000,000+
Antillean Gold Telegram (original version) : 100,000+
AT&T Visual Voicemail : 10,000,000+
GMX Mail : 10,000,000+
Omlet Chat : 10,000,000+
My Vodacom SA : 5,000,000+
Microsoft Edge : 5,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Calls & Text by Mo+ : 5,000,000+
free 

If we remove all the apps from Communication genre that have 100 millions or more installation, then the average will come down nearly ten times!

In [28]:
under_100_m = []
for app in android_final:
    n_install = app[5]
    n_install = n_install.replace(',', '')
    n_install = n_install.replace('+', '')
    n_install = float(n_install)
    if app[1] == 'COMMUNICATION' and n_install < 100000000:
        under_100_m.append(n_install)
under_avg_install = sum(under_100_m) / len(under_100_m)
print(under_avg_install)

3603485.3884615386


We see the same pattern for the video players category, which is the runner-up with 24,727,872 installs. The market is dominated by apps like Youtube, Google Play Movies & TV, or MX Player. The pattern is repeated for social apps (where we have giants like Facebook, Instagram, Google+, etc.), photography apps (Google Photos and other popular photo editors), or productivity apps (Microsoft Word, Dropbox, Google Calendar, Evernote, etc.).

Again, the main concern is that these app genres might seem more popular than they really are. Moreover, these niches seem to be dominated by a few giants who are hard to compete against.

The game genre seems pretty popular, but previously we found out this part of the market seems a bit saturated, so we'd like to come up with a different app recommendation if possible.

The books and reference genre looks fairly popular as well, with an average number of installs of 8,767,811. It's interesting to explore this in more depth, since we found this genre has some potential to work well on the App Store, and our aim is to recommend an app genre that shows potential for being profitable on both the App Store and Google Play Store.

Lets take a look at the apps of this genre and their number of installations.

In [29]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

The book and reference genre includes a variety of apps: software for processing and reading ebooks, various collections of libraries, dictionaries, tutorials on programming or languages, etc and this market still shows potential.

Let's try to get some app ideas based on the kind of apps that are somewhere in the middle in terms of popularity (between 1,000,000 and 100,000,000 downloads).

In [30]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+' 
                                            or app[5] == '5,000,000+' 
                                            or app[5] == '10,000,000+' 
                                            or app[5] == '50,000,000+'
                                            or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Aldiko Book Reader : 10,000,000+
Wattpad 📖 Free Books : 100,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim -

This niche seems to be dominated by software for processing and reading ebooks, as well as various collections of libraries and dictionaries, so it's probably not a good idea to build similar apps since there'll be some significant competition.

We also notice there are quite a few apps built around the same book (in this case the religious book 'Al Quran'), which suggests that building an app around a popular book can be profitable. It seems that taking a popular book and turning it into an app could be profitable for both the Google Play Store and the Apple App Store market.

However, it looks like the market is already full of libraries, so we need to add some special features besides the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.

# Conclusions

In this project, we have analyzed data from Google Play Store and Apple App Store with the aim to find out and recommend the best app that can be built and can be profitable and successful on both the markets in the long run.

We concluded that choosing a popular book and turning it into an app could be profitable for both the Google Play Store and the App Store markets. However, for it to be succesful, it is recommended that we add some extra features in the app alongside with the actual book.